### PR TITLE
[Communication] - Identifiers - Fixing ObjC interface

### PR DESCRIPTION
**Pull request by JoshuaLai**

### CODE REVIEW

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift [hunk 1]
```diff
@@ -28,24 +28,21 @@
...
         self.environmentValue = environmentValue
     }
 
     static func fromModel(environmentModel: CommunicationCloudEnvironmentModel) -> CommunicationCloudEnvironment {
         return CommunicationCloudEnvironment(environmentValue: environmentModel.requestString);
     }
-}
 
-// CommunicationCloudEnvironment Extension
-extension CommunicationCloudEnvironment: Equatable {
```
Comment (sacheu) [inline sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:48]: if you remove `CommunicationCloudEnvironment: Equatable `, will the == func gets called ?
when we do 'aCommunicationCloudEnvironment == anotherCommunicationCloudEnvironment`?

**Review by sacheu — CHANGES_REQUESTED** [review]
I have a question about removing Equatable from CommunicationCloudEnvironment

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift (hunk 1 above)
Comment (JoshuaLai) [reply sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:48]: The implementation of `==` in swift will not be effected. But this implementation will not exist for ObjC

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift [hunk 2]
```diff
@@ -28,24 +28,21 @@
...
     public static let Public = CommunicationCloudEnvironment(environmentValue: "public")
     public static let Dod = CommunicationCloudEnvironment(environmentValue: "dod")
     public static let Gcch = CommunicationCloudEnvironment(environmentValue: "gcch")
 
-    init(environmentValue: String) {
+    public init(environmentValue: String) {
         self.environmentValue = environmentValue
     }
 
     static func fromModel(environmentModel: CommunicationCloudEnvironmentModel) -> CommunicationCloudEnvironment {
```
Comment (JoshuaLai) [inline sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:42]: @RezaJooyandeh Can you help explain a bit about who is going to call this method?

Comment (RezaJooyandeh) [reply sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:42]: This method is to be used by communication libraries like chat when they are trying to deserialize an instance of `CommunicationCloudEnvironmentModel` that has been in a response.

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift (hunk 1 above)
Comment (sacheu) [reply sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:48]: XCode gives us a warning saying we should implement `isEqual`

image

Comment (sacheu) [reply sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:48]: And should we move the static func out for PhoneNumberIdentifier as well?

**Review by sacheu — APPROVED** [review]
PR looks good.

File: sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift (hunk 1 above)
Comment (sacheu) [reply sdk/communication/AzureCommunication/Source/CommunicationCloudEnvironment.swift:48]: My mistake. We should leave the == as is.

File: sdk/communication/AzureCommunication/Source/Identifiers.swift [hunk 3]
```diff
@@ -116,13 +116,20 @@ import Foundation
...
      */
-    public init(userId: String, isAnonymous: Bool = false, identifier: String? = nil, cloudEnvironment: CommunicationCloudEnvironment = CommunicationCloudEnvironment.Public) {
+    public init(userId: String, isAnonymous: Bool = false, identifier: String? = nil, cloudEnvironment: CommunicationCloudEnvironment = .Public) {
         self.identifier = identifier
         self.userId = userId
         self.isAnonymous = isAnonymous
         self.cloudEnviroment = cloudEnvironment
     }
 
+    public init(userId: String, isAnonymous: Bool) {
```
Comment (JoshuaLai) [inline sdk/communication/AzureCommunication/Source/Identifiers.swift:126]: @raosanat I added a new init to fill the roll of default parameters to make it a little easier in Obj-C

**Review by sacheu — APPROVED** [review]
PR looks good.

**Pull request merged**